### PR TITLE
  Add assignment name and content hash to dsc.config file

### PR DESCRIPTION
**Pull request by amitsara**
## Description

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.
- [ ] If needed, Version of the Extension was bumped in the misc/manifest.xml file.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/Guest-Configuration-Extension/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: main/utils.go [hunk 1]
```diff
@@ -269,3 +270,65 @@ func getStdPipesAndTelemetry(lg ExtensionLogger, logDir string, runErr error) {
...
+        os.MkdirAll(dscConfigFolderPath, os.ModePerm)
+
+        // New dsc.config file
+        firstAssignmentString := "{\"Assignments\": [ { \"name\": \"" + assignmentName + "\", \"contentHash\": \"" + contentHash + "\" }]}"
+        var firstAssignment map[string]interface{}
+        json.Unmarshal([]byte(firstAssignmentString ), &firstAssignment)
+        firstAssignmentJson, _ := json.Marshal(firstAssignment)
+        err = ioutil.WriteFile(dscConfigFilePath, firstAssignmentJson, 0644)
+        if err != nil {
+            fmt.Println(err)
```
Comment (BerheAbrha) [inline main/utils.go]: >Println [](start = 16, length = 7)

return 'err' here ?  It should fail if the writing content hash is not working.

File: main/cmds.go [hunk 2]
```diff
@@ -72,6 +72,9 @@ func enable(lg ExtensionLogger, hEnv vmextension.HandlerEnvironment, seqNum int)
 	//	return errors.Wrap(err, "failed to parse version string")
 	//}
 
+	// Add Assignment name and content hash in dsc.config file.
+	updateAssignment(cfg.publicSettings.AssignmentName, cfg.publicSettings.ContentHash)
```
Comment (BerheAbrha) [inline main/cmds.go]: I think we need to surface the error back to user. similar to the other calls

File: main/utils.go [hunk 3]
```diff
@@ -269,3 +270,65 @@ func getStdPipesAndTelemetry(lg ExtensionLogger, logDir string, runErr error) {
...
+    if err != nil {
+        dscConfigFolderPath := "/var/lib/GuestConfig/dsc"
+        os.MkdirAll(dscConfigFolderPath, os.ModePerm)
+
+        // New dsc.config file
+        firstAssignmentString := "{\"Assignments\": [ { \"name\": \"" + assignmentName + "\", \"contentHash\": \"" + contentHash + "\" }]}"
+        var firstAssignment map[string]interface{}
+        json.Unmarshal([]byte(firstAssignmentString ), &firstAssignment)
+        firstAssignmentJson, _ := json.Marshal(firstAssignment)
+        err = ioutil.WriteFile(dscConfigFilePath, firstAssignmentJson, 0644)
```
Comment (BerheAbrha) [inline main/utils.go]: >0644 [](start = 71, length = 4)

what is this number .. could you define a const for it ?

Comment (amitsara) [reply main/utils.go]: this means owner have permission to read & write and other can only read. defined a const for this

---
In reply to: [296053478](https://github.com/Azure/Guest-Configuration-Extension/pull/87#discussion_r296053478) [](ancestors = 296053478)

**Pull request merged**